clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerPartitionAssignor.java

```diff
@@ -45,17 +45,18 @@
      * and can be leveraged in {@link #assign(Cluster, GroupSubscription)} ((e.g. local host/rack information)
      *
      * @param topics Topics subscribed to through {@link org.apache.kafka.clients.consumer.KafkaConsumer#subscribe(java.util.Collection)}
-     *               and variants
+     *               and variants.
+     * @param ownedPartitions Modifiable list of owned partitions that will be included in the Subscription sent to the leader.
      * @return nullable subscription user data
      */
-    default ByteBuffer subscriptionUserData(Set<String> topics) {
+    default ByteBuffer subscriptionUserData(Set<String> topics, List<TopicPartition> ownedPartitions) {
```

Comment:
Since it is approaching the release deadline and we've not done the perf analysis on whether or not this would incur big benefits on streams yet, while on the other hand we do want to complete the current scope of KIP-429 code changes, wdyt?